applications/Unity.GrantManager/modules/Unity.Payments/src/Unity.Payments.Application/Integration/Cas/InvoiceService.cs

```diff
@@ -0,0 +1,325 @@
+﻿using Volo.Abp;
+using RestSharp;
+using RestSharp.Authenticators;
+using System.Net;
+using System.Threading.Tasks;
+using System.Text.Json;
+using System;
+using Unity.Payments.Integrations.Cas;
+using Unity.Payments.Integrations.Http;
+using Unity.Payments.Integration.Http;
+using Volo.Abp.Application.Services;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Logging;
+using System.Collections.Generic;
+using System.Net.Http;
+using Unity.Payments.Enums;
+using Unity.Payments.Domain.Suppliers;
+using Unity.Payments.PaymentConfigurations;
+using Unity.Payments.Domain.PaymentRequests;
+using System.Runtime.Intrinsics.X86;
+using System.Linq;
+using Volo.Abp.Domain.Repositories;
+
+
+namespace Unity.Payments.Integration.Cas
+{
+    [IntegrationService]
+    public class InvoiceService : ApplicationService, IInvoiceService
+    {
+
+        private readonly IPaymentRequestRepository _iPaymentRequestRepository;
+        private readonly IResilientHttpRequest _resilientRestClient;
+        private readonly ISiteRepository _iSiteRepository;
+        private readonly ISupplierRepository _iSupplierRepository;
+        private readonly IOptions<CasClientOptions> _casClientOptions;
+        private IPaymentConfigurationAppService _paymentConfigurationAppService;
+        private readonly RestClient _restClient;
+        private const string OAUTH_PATH = "oauth/token";
+        private const string CFS_APINVOICE = "cfs/apinvoice";
+
+        public readonly Dictionary<int, string> CASPaymentGroup = new Dictionary<int, string>
+        {
+            { (int)PaymentGroup.EFT, "GEN EFT" },
+            { (int)PaymentGroup.Cheque, "GEN CHQ" }
+        };
+
+        public InvoiceService(
+            IPaymentRequestRepository paymentRequestRepository,
+            IPaymentConfigurationAppService paymentConfigurationAppService,
+            IResilientHttpRequest resilientHttpRequest,
+            IOptions<CasClientOptions> casClientOptions,
+            ISupplierRepository iSupplierRepository,
+            ISiteRepository iSiteRepository,
+            RestClient restClient)
+        {
+            _iPaymentRequestRepository = paymentRequestRepository;
+            _paymentConfigurationAppService = paymentConfigurationAppService;
+            _resilientRestClient = resilientHttpRequest;
+            _casClientOptions = casClientOptions;
+            _iSupplierRepository = iSupplierRepository;
+            _iSiteRepository = iSiteRepository;
+            _restClient = restClient;
+        }
+
+        private async Task<Invoice?> InitializeCASInvoice(PaymentRequest paymentRequest,
```

Comment:
Look at the best practices for application services - as above, trying to be good with implementation / best practices with all the new modules

https://docs.abp.io/en/abp/latest/Best-Practices/Application-Services

In particular:
Application Service Implementation
Do develop the application layer completely independent from the web layer.
Do implement application service interfaces in the application layer.
Do use the naming convention. Ex: Create ProductAppService class for the IProductAppService interface.
Do inherit from the ApplicationService base class.
Do make all public methods virtual, so developers may inherit and override them.
Do not make private methods. Instead make them protected virtual, so developers may inherit and override them.